### PR TITLE
fix: always return primary keys of associated records

### PR DESCRIPTION
**Pull request by ghusse**
- ✅ &nbsp; Tested with the model that shown the error in Forest Admin
- ✅ &nbsp; Tested with the model from https://app.clickup.com/t/8ppqbf

## Pull Request checklist:

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [x] Create automatic tests
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

File: test/services/query-builder.test.js [hunk 1]
```diff
@@ -70,20 +71,82 @@ describe('services > query-builder', () => {
...
+
+      it('should always include the target key and primary key even if not specified', () => {
+        expect.assertions(1);
+        const { model, queryBuilder } = setup();
+
+        const includes = queryBuilder.getIncludes(model, ['user.name', 'user.id', 'user.badField']);
+
+        expect(includes).toStrictEqual([{
+          as: 'userAccessor',
+          attributes: ['p_id', 'uid', 'name'],
```
Comment (arnaudbesnier) [inline test/services/query-builder.test.js:149]: Kind of same question as above ☝️ 

I don't get why the target `'uid'` is set here.
Don't we just need:
- the primary key of the record (for ember data)
- the displayed field for the data display in the user interface.
?

File: test/services/query-builder.test.js [hunk 2]
```diff
@@ -70,20 +71,82 @@ describe('services > query-builder', () => {
...
+      it('should always include the source key and primary key even if not specified', () => {
         expect.assertions(1);
         const { model, queryBuilder } = setup();
 
         const includes = queryBuilder.getIncludes(model, ['user.name', 'user.id', 'user.badField']);
 
         expect(includes).toStrictEqual([{
           as: 'userAccessor',
-          attributes: ['uid', 'name'],
+          attributes: ['p_id', 'uid', 'name'],
```
Comment (arnaudbesnier) [inline test/services/query-builder.test.js:87]: Why do we need the source key?
The primary key + display field should be enough right?
Is it for join purpose?

File: test/services/query-builder.test.js [hunk 3]
```diff
@@ -31,7 +31,9 @@ describe('services > query-builder', () => {
     describe('with a hasOne relationship', () => {
       function setup() {
         const target = {
+          primaryKeyAttributes: ['pid'],
```
Comment (arnaudbesnier) [inline test/services/query-builder.test.js:34]: So you changed the primary key from `id` to `pid` in this test, right?
Is that really necessary? `id` as a primary key is simpler to understand.

File: test/services/query-builder.test.js (hunk 1 above)
Comment (ghusse) [reply test/services/query-builder.test.js]: Yes, I changed the behavior to do this instead of returning the source/target fields plus the primary key

**Review by arnaudbesnier — APPROVED** [review]

**Pull request merged**